x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/optimizer/OptimizerRules.java

```diff
@@ -1030,9 +1029,7 @@ protected LogicalPlan rule(Filter filter) {
                     return filter.child();
                 }
                 if (FALSE.equals(condition) || Expressions.isNull(condition)) {
-                    //TODO: re-visit this branch when it's decided if EQL needs a LocalRelation-like class
-                    //return new LocalRelation(filter.source(), new EmptyExecutable(filter.output()));
-                    throw new RuleExecutionException("Does not know how to handle a local relation");
+                    return nonMatchingFilter(filter);
```

Comment:
Thanks for this fix! Looks like this resolves
https://github.com/elastic/elasticsearch/issues/53237

Reply:
Not entirely - there is still some work to be done there mainly on returning the results from the empty executable however I didn't want to add that into this PR.